### PR TITLE
Allow interpreting Int32 as Long when reading BSON values

### PR DESCRIPTION
**Pull request by sebaciv**

### CODE REVIEW

**Review by ghik — COMMENTED** [review]
Can you also make it _write_ an INT32 when a `Long` value fits in an `Int`?

**Comment by sebaciv** [comment]
I suppose it can be done, but I don't understand why? That would change DB representation of existing data after some update operation. IMO it's better if the type in DB corresponds to the type declared in the code. Change introduced by this PR is only for backward compatibility.

File: commons-mongo/jvm/src/test/scala/com/avsystem/commons/mongo/SomethingPlain.scala [hunk 1]
```diff
@@ -80,3 +80,8 @@ object SomethingComplex {
 
   implicit val codec: GenCodec[SomethingComplex] = GenCodec.materialize
 }
+
+case class SomethingLong(value: Long)
+object SomethingLong {
```
Comment (ghik) [inline commons-mongo/jvm/src/test/scala/com/avsystem/commons/mongo/SomethingPlain.scala:85]: `HasGenCodec` ?

File: commons-mongo/jvm/src/main/scala/com/avsystem/commons/mongo/BsonReaderInput.scala [hunk 2]
```diff
@@ -16,7 +16,10 @@ class BsonReaderInput(br: BsonReader, override val legacyOptionEncoding: Boolean
   override def readString(): String = br.readString()
   override def readBoolean(): Boolean = br.readBoolean()
   override def readInt(): Int = br.readInt32()
-  override def readLong(): Long = br.readInt64()
+  override def readLong(): Long = {
```
Comment (ghik) [inline commons-mongo/jvm/src/main/scala/com/avsystem/commons/mongo/BsonReaderInput.scala:19]: Cosmetic: pls use pattern matching on `bsonType`

**Review by ghik — APPROVED** [review]

**Pull request merged**